src/test/ensemble_flow_custom_node_tests.cpp

```diff
@@ -4928,38 +4896,33 @@ TEST_F(EnsembleFlowCustomNodePipelineExecutionTest, MultipleDeinitializeCallsOnR
         {"custom_node_3", {{customNodeOutputName, pipelineOutputName}}}};
 
     ASSERT_EQ(factory.createDefinition("my_new_pipeline", info, connections, manager), StatusCode::OK);
+    waitForOVMSResourcesCleanup(manager);
+    ASSERT_EQ(manager.getResourcesSize(), 3);
 
     factory.retireOtherThan({}, manager);
-    // Each custom node has effectively 1 internalManager initialized, because they use same library instance
-    // in order to count whether deinitialize has been called expected number of times
-    ASSERT_EQ(LibraryCountDeinitialize::deinitializeCounter, 3);
```

Comment:
Keeping track of resources count in ModelManager is nice for test as additional check. However with this library we could also verify if there was no additional deinitialize calls.